sky/spot/recovery_strategy.py

```diff
@@ -512,7 +546,8 @@ def recover(self) -> float:
                                              region=launched_region)
                 }
                 # Not using self.launch to avoid the retry until up logic.
-                job_submitted_at = self._launch(raise_on_failure=False)
+                job_submitted_at = self._launch(
```

Comment:
recover() may crash due to the underlying _launch() raising some non-retryable errors like failed prechecks. Previously recover() will not crash but instead receive None = _launch(…) even if non-retryable errors happened.

Is that ok? Have we tested this code path?